src/test/java/com/uid2/admin/vertx/EncryptionKeyServiceTest.java

```diff
@@ -154,11 +188,131 @@ void rotateMasterKey(Vertx vertx, VertxTestContext testContext) throws Exception
+        vertx.deployVerticle(verticle, testContext.succeeding(id -> testContext.completeNow()));
+        // set it to be key creation timestamp + 100days so that we can create expired keys [UID2-599]
+        when(clock.now()).thenReturn(Instant.ofEpochMilli(KEY_CREATE_TIME_IN_MILLI + TEN_DAYS_IN_MILLI * 10));
+        fakeAuth(Role.SECRET_MANAGER);
+
+        final EncryptionKey[] keys = {
+                new EncryptionKey(11, null, Instant.ofEpochMilli(KEY_CREATE_TIME_IN_MILLI), Instant.ofEpochMilli(KEY_ACTIVATE_TIME_IN_MILLI), Instant.ofEpochMilli(KEY_EXPIRE_TIME_IN_MILLI), -1),
+        };
+
+        setEncryptionKeys(MAX_KEY_ID, keys);
+
+        post(vertx, "api/key/rotate_master?min_age_seconds=100", "", ar -> {
+            assertTrue(ar.succeeded());
+            HttpResponse response = ar.result();
+            assertEquals(200, response.statusCode());
+            checkRotatedKeyResponse(MAX_KEY_ID+1, new int[] { -1, },
+                    MASTER_KEY_ACTIVATES_IN_SECONDS, MASTER_KEY_EXPIRES_AFTER_SECONDS,
+                    response.bodyAsJsonArray().stream().toArray());
+            try {
+                verify(encryptionKeyStoreWriter, times(1)).upload(collectionOfSize(1), eq(MAX_KEY_ID+1));
+            } catch (Exception ex) {
+                fail(ex);
+            }
+            testContext.completeNow();
+        });
+    }
+
+    @Test
+    void filterOutMasterKeysOverCutoffTimeWithFlagOff(Vertx vertx, VertxTestContext testContext) throws Exception {
+        // set it to be key creation timestamp + 100days so that we can create expired keys [UID2-599]
+        when(clock.now()).thenReturn(Instant.ofEpochMilli(KEY_CREATE_TIME_IN_MILLI + TEN_DAYS_IN_MILLI * 10));
+        fakeAuth(Role.SECRET_MANAGER);
+
+        final EncryptionKey[] keys = {
+                new EncryptionKey(11, null, Instant.ofEpochMilli(KEY_CREATE_TIME_IN_MILLI), Instant.ofEpochMilli(KEY_ACTIVATE_TIME_IN_MILLI), Instant.ofEpochMilli(KEY_EXPIRE_TIME_IN_MILLI), -1),
+        };
+
+        setEncryptionKeys(MAX_KEY_ID, keys);
+
+        post(vertx, "api/key/rotate_master?min_age_seconds=100", "", ar -> {
+            assertTrue(ar.succeeded());
+            HttpResponse response = ar.result();
+            assertEquals(200, response.statusCode());
+            checkRotatedKeyResponse(MAX_KEY_ID+1, new int[] { -1, },
+                    MASTER_KEY_ACTIVATES_IN_SECONDS, MASTER_KEY_EXPIRES_AFTER_SECONDS,
+                    response.bodyAsJsonArray().stream().toArray());
+            try {
+                verify(encryptionKeyStoreWriter, times(1)).upload(collectionOfSize(2), eq(MAX_KEY_ID+1));
+            } catch (Exception ex) {
+                fail(ex);
+            }
+            testContext.completeNow();
+        });
+    }
+
+    @Test
+    void filterOutRefreshKeysOverCutoffTimeWithFlagOn(Vertx vertx, VertxTestContext testContext) throws Exception {
+        this.config.put("filter_key_over_cut_off_days", true);
```

Comment:
This part is nasty I know as it repeated 3 times, I should've put it in a function :( 

But we will be removing this flag later on if everything is okay. So I suggest to keep it this way and remove it later on.

Reply:
Fine for code we know we will remove.